src/libs/colorpicker.h

```diff
@@ -43,12 +56,48 @@ typedef struct dt_colorpicker_sample_t
   dt_aligned_pixel_t picked_color_lab_max;
 
   /** The GUI elements */
+  // FIXME: these may be only for primary and live pickers -- keep local to colorpicker.c if so?
   GtkWidget *container;
   GtkWidget *color_patch;
   GtkWidget *output_label;
   GdkRGBA rgb;
 } dt_colorpicker_sample_t;
 
+// FIXME: should this be in a colorspace-specific utility header, e.g. iop_profile or colorspaces or colorspaces_inline_conversions
+static inline gboolean dt_lib_colorpicker_convert_color_space(const GdkRGBA *restrict sample, GdkRGBA *restrict color)
+{
+  // RGB values are relative to the histogram color profile
+  // we need to adapt them to display profile so color look right
+  // Note : dt_ioppr_set_pipe_output_profile_info sets a non-handled output profile to sRGB by default
+  // meaning that this conversion is wrong for fancy-pants LUT-based display profiles.
+
+  dt_iop_order_iccprofile_info_t *histogram_profile = dt_ioppr_get_histogram_profile_info(darktable.develop);
+  dt_iop_order_iccprofile_info_t *display_profile = dt_ioppr_get_pipe_output_profile_info(darktable.develop->pipe);
```

Comment:
seems like this causes CI to bail because `colorpicker.h` does not include `common/iop_profile.h`

Reply:
Yes! I had to take a bit of time away, but got back to this and have a fix now. This whole bit of code is funny:

- `libs/colorpicker.c` uses inline `dt_ioppr_rgb_matrix_to_xyz` and `dt_ioppr_xyz_to_rgb_matrix` rather than the (still very fast for matrix profiles) `dt_ioppr_transform_image_colorspace_rgb`.
- I copied this to do the color math (histogram -> display profile) for displaying the picker overlay, but moved the code to `libs/colorpicker.h`, and was missing that `common/iop_profile.h` include, hence the build error.
- This is easy to fix, and indeed, the single pixel rgb -> rgb code really could belong in `common/iop_profile.h` as general purpose code.
- While doing this specialized single pixel conversion is stunningly fast (10x faster than calling `dt_ioppr_transform_image_colorspace_rgb`), either is ludicrously quick (comparison between processing 100,000+ samples/sec. or 1,000,000+ samples/sec. when we're only processing a single sample).
- So does it make sense to just use `dt_ioppr_transform_image_colorspace_rgb()` and take the 10x speed hit and keep the code a bit simpler...?
- An advantage of `dt_ioppr_transform_image_colorspace_rgb()` is that it can handle non-matrix profiles via LCMS. Histogram profile must be matrix, but display profile might not be, so this could be nice...